### PR TITLE
add (some of) Kafka

### PR DESCRIPTION
**Pull request by SethTisue**
using Enno's fork that adds an sbt build /cc @ennru

### CODE REVIEW

**Comment by SethTisue** [comment]
trial run queued: 
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3797/

**Comment by ennru** [comment]
That went surprisingly well:
```
[kafka] [info] Tests: succeeded 415, failed 10, canceled 0, ignored 0, pending 0
[kafka] [info] *** 10 TESTS FAILED ***
[kafka] [error] Failed tests:
[kafka] [error] 	kafka.api.UserQuotaTest
[kafka] [error] 	kafka.server.FetchRequestTest
[kafka] [error] 	kafka.server.AbstractCreateTopicsRequestTest
[kafka] [error] 	kafka.admin.ConsumerGroupCommandTest
[kafka] [error] 	kafka.api.MetricsTest
```

I'm not sure when I'll have the time to understand why these tests fail, should we exclude those for now?

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3804/

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3806/

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3808/

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3810/

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3812/

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3814/

**Comment by SethTisue** [comment]
@ennru in your build.sbt I see `ThisBuild / scalaVersion := "2.11.12"` but we are no longer adding projects to the Scala 2.11 community build, this PR targets 2.12.x. dbuild overrides the Scala version, so it isn't strictly necessary that you change it, but I just wanted to let you know that latest-2.12.x is the version you should be testing if you do any testing yourself on your branch

**Comment by SethTisue** [comment]
@ennru the latest run log is a bit mysterious — I think a test (I'm not sure which one) is going haywire and killing the JVM entirely rather than merely failing the test. the same thing happens when I check out your branch and do `++2.12.7!` followed by `core/test`

**Comment by SethTisue** [comment]
@ennru I see in the Kafka repo readme that there is a distinction between "unit" and "integration" tests. perhaps we should only be running the "unit" tests?

**Comment by ennru** [comment]
Ok, I'll see how we can restrict to only run the unit tests.
I'll change to 2.12 as default, as well. The Gradle build supports both 2.11 and 2.12 and adds extra compiler flags for 2.12.

**Comment by ennru** [comment]
I've updated my clone/branch to exclude a couple of tests.

The JUnit category for Integration tests is mostly used in the Kafka Streams module which the sbt build doesn't use.

**Comment by ennru** [comment]
And I pull in the latest changes from Kafka trunk.

**Comment by SethTisue** [comment]
(I'm back from vacation, and also done unfreezing the community build and bringing it up to date)

let's have Jenkins try this again (with latest changes from 2.12.x branch merged): 
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3939/ (404 for a while til queue clears)

**Comment by SethTisue** [comment]
```
[kafka] [info] *** 60 TESTS FAILED ***
[kafka] [error] Failed tests:
[kafka] [error] 	kafka.api.SaslMultiMechanismConsumerTest
[kafka] [error] 	kafka.api.SaslSslConsumerTest
[kafka] [error] 	kafka.api.SaslGssapiSslEndToEndAuthorizationTest
[kafka] [error] 	kafka.zookeeper.ZooKeeperClientTest
[kafka] [error] 	kafka.server.MultipleListenersWithDefaultJaasContextTest
[kafka] [error] 	kafka.api.SaslSslAdminClientIntegrationTest
[kafka] [error] 	kafka.api.SaslPlaintextConsumerTest
[kafka] [error] 	kafka.server.ServerShutdownTest
[kafka] [error] 	kafka.server.MultipleListenersWithAdditionalJaasContextTest
[kafka] [error] 	kafka.server.GssapiAuthenticationTest
```

**Comment by ennru** [comment]
Hm, those are intended to be excluded on source level in sbt. Can you recommend a stronger way to exclude them?

File: configs/community.dbuild [hunk 1]
```diff
@@ -1916,6 +1916,18 @@ build += {
...
 
+  ${vars.base} {
+    name: "kafka"
+    uri:  ${vars.uris.kafka-uri}
+    extra.commands: ${vars.default-commands} [
+      // "I'm not sure when I'll have the time to understand why these tests fail",
+      // says Enno. me either (Seth).  note that some of these actually failed, and
+      // then the rest stopped even compiling once we excluded
+      // AbstractCreateTopicsRequestTest.scala
+      "set core / Test / unmanagedSources / excludeFilter := HiddenFileFilter || \"UserQuotaTest.scala\" || \"FetchRequestTest.scala\" || \"AbstractCreateTopicsRequestTest.scala\" || \"ConsumerGroupCommandTest.scala\" || \"MetricsTest.scala\" || \"DeleteConsumerGroupsTest.scala\" || \"DescribeConsumerGroupTest.scala\" || \"ResetConsumerGroupOffsetTest.scala\" || \"ListConsumerGroupTest.scala\" || \"CreateTopicsRequestTest.scala\" || \"CreateTopicsRequestWithPolicyTest.scala\""
```
Comment (ennru) [inline configs/community.dbuild]: This overwrites the settings I added directly to the `build.sbt` in my `build-with-sbt` branch. Please remove this...

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/3955/ (404 til Jenkins wakes up)

**Comment by ennru** [comment]
Looks like it went through.

**Comment by SethTisue** [comment]
yay! thanks for doing the heavy lifting here

**Comment by SethTisue** [comment]
I'll highlight this in https://github.com/scala/scala-lang/pull/982

**Comment by ennru** [comment]
I'll have a look if other parts of the Kafka source should become part of the community build.

**Pull request merged**